### PR TITLE
Backend pokemon api requests

### PR DESCRIPTION
**Pull request by orogeny**
### Finally, the server API code.

I've looked at filling the Card's description with species "flavor_text_entries" but there seems to be duplicate text entries (discriminated by 'color') - not insurmountable and probably better with a plan.

I've also moved all the pokemon specific code into a "pokemon" folder and just copied/pasted Jenny's changes to the card.types.ts file so there *shouldn't* be any conflict.

If you could have a look and any feedback appreciated.

Cheers, Paul.

## What in the PR

# GET http://localhost:8080/api/pokemon

returns an array of {id: x, name: <pokemon character name>} objects

# GET http://localhost:8080/api/pokemon/{id}

returns a Card for that pokeman

### CODE REVIEW

File: server/pokemon/toCard.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import { Card, STAT_NAME } from "../card/card.types";
+import { PokemonDto } from "./pokemon.types";
+
+function toCard(dto: PokemonDto) {
+  const stats = STAT_NAME.map((statName, index) => {
+    const value =
+      statName in dto
+        ? // @ts-ignore
```
Comment (jennysivapalan) [inline server/pokemon/toCard.ts:8]: what does this do?

**Review by jennysivapalan — APPROVED** [review]
Nice looks great Paul! We can follow up with updating the tests to not fetch the API direct next week if we want to.

File: server/pokemon/toCard.ts (hunk 1 above)
Comment (orogeny) [reply server/pokemon/toCard.ts:8]: Should have mentioned that, Jenny.  it's from a hell of a lot of Googling !

The toCard() function looks for stat (STAT_NAME) values in the Pokemon's stats AND in the main object too (weight, height and hp).  

To check a property is on the object you do

if (property_name in obj) value = obj["property_name"];

Even though you've done the check, Typescript still can't handle the truth, so you have to tell it to ignore the check.

Comment (jennysivapalan) [reply server/pokemon/toCard.ts:8]: ahh okay thanks for the explanation!

**Pull request merged**